astropy/convolution/convolve.py

```diff
@@ -199,110 +281,109 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         # *kernel* doesn't support NaN interpolation, so instead we just fill it
         kernel_internal = kernel.filled(fill_value)
 
-    # Mark the NaN values so we can replace them later if interpolate_nan is
-    # not set
-    if preserve_nan:
-        badvals = np.isnan(array_internal)
-
-    if nan_treatment == 'fill':
-        initially_nan = np.isnan(array_internal)
-        array_internal[initially_nan] = fill_value
+    # NaN interpolation significantly slows down the C convolution
+    # compuatation. Since nan_treatment = 'interpolate', is the default
+    # check whether it is even needed, if not, don't interpolate.
+    # NB: np.isnan(array_internal.sum()) is fatser than np.isnan(array_internal).any()
```

Comment:
faster.

Have you checked performance of `np.count_nonzero(np.isnan(array_internal))`? It might just be that the per-pixel checking of isnan is the performance bottleneck, in which case `array_internal.sum()` (or maybe `.max()`?) is going to be the fastest.

Reply:
I did, and you're correct, the perf. bottleneck is having to check each pix for `NaN`s. Which is why I went for `sum()`, and then only check the result.  I didn't think of using `max()`, but thinking about it now, `sum()` has fewer operations and is vectorizable.